src/utils.rs

```diff
@@ -193,20 +173,37 @@ mod tests {
             assert_eq!(phrase, TEST_MNEMONIC)
         });
     }
+
     #[test]
-    fn derive_account_by_index() {
+    fn write_wallet() {
         with_tmp_folder(|tmp_folder| {
-            // initialize a wallet
             let wallet_path = tmp_folder.join("wallet.json");
-            save_dummy_wallet_file(&wallet_path);
-            // derive account with account index 0
-            let account_ix = 0;
-            let private_key = derive_account(&wallet_path, account_ix, TEST_PASSWORD).unwrap();
-            assert_eq!(
-                private_key.to_string(),
-                "961bf9754dd036dd13b1d543b3c0f74062bc4ac668ea89d38ce8d712c591f5cf"
-            )
-        });
+            write_wallet_from_mnemonic_and_password(&wallet_path, TEST_MNEMONIC, TEST_PASSWORD)
+                .unwrap();
+            load_wallet(&wallet_path).unwrap();
+        })
```

Comment:
Do we need to clean up the file after each test so it doesn't interfere with the test the second time it's run?

```suggestion
            std::fs::remove_file(wallet_path).unwrap();
        })
```

Reply:
We only need to manually clean up the new `handle_relative_path_argument` test, as we're specifically testing a path relative to where the command is invoked. For all other tests, we use the `with_tmp_folder` function that is automatically cleaned up after the test completes.